### PR TITLE
feat: implement checkout order items with shipping options

### PR DESCRIPTION
**Pull request by sss-bct**

### CODE REVIEW

File: components/common/ProductItem/ProductItem.spec.tsx [hunk 1]
```diff
@@ -60,3 +60,32 @@ describe('[component] - ProductItem with Price Label', () => {
...
+  }
+
+  it('should not show details when no product label or oprions is present', () => {
+    setup()
+
+    const detailsElement = screen.queryByText(/details/i)
+
+    expect(detailsElement).not.toBeInTheDocument()
+  })
+})
```
Comment (kibo-sushant) [inline components/common/ProductItem/ProductItem.spec.tsx:91]: **As part of 'should render component', We should test the following things**
the title is displayed or not
Qty is displayed or not
Price is displayed or not
Mocked Price is displayed or not
Mocked productOptionListMock is displayed or not

**Missing TestCases**
should display details option when there is either Qty or Price or Options
should expand details section when user clicks on shevron icon
should shrink details section when user clicks on shevron icon again

**We can delete following testcases**
it('should render mock component'
describe('[component] - ProductItem with Price Label'

File: components/common/ProductItemList/ProductItemList.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          {...item}
+          key={`${item.name}-${index}`}
+          data-testid="product-item"
+        ></ProductItem>
+      ))}
+    </Stack>
+  )
+}
+
+export default ProductItemList
```
Comment (kibo-sushant) [inline components/common/ProductItemList/ProductItemList.tsx:31]: **Add spacing between unrelated code**
const { items } = props
return (

File: components/common/ProductItem/ProductItem.stories.tsx [hunk 3]
```diff
@@ -67,31 +67,45 @@ const Template: ComponentStory<typeof ProductItem> = (args) => (
...
+  qty: cartItem?.quantity,
+  price: '$' + (cartItem.product?.price?.price || 0).toString(),
+  salePrice: '$' + (cartItem.product?.price?.salePrice || 0).toString(),
+}
+
+export const WithoutDetailOption = TemplateWithQtyLabel.bind({})
+Common.args.options = []
+WithoutDetailOption.args = {
+  ...Common.args,
+}
```
Comment (kibo-sushant) [inline components/common/ProductItem/ProductItem.stories.tsx:111]: **Incorrect argTypes**
 argTypes: {
    size: { control: 'select' },
    fontWeight: { control: 'radio' },
  }

File: components/checkout/Shipping/OrderItems/ShippingMethod.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  return (
+    <Box data-testid="shipping-method">
+      {shipItems?.length ? <ShipItemList /> : ''}
+
+      {pickupItems?.length ? <PickupItemList /> : ''}
+    </Box>
+  )
+}
+
+export default ShippingMethod
```
Comment (kibo-sushant) [inline components/checkout/Shipping/OrderItems/ShippingMethod.tsx:91]: **Using Font and FontFamily**
fontSize: 18
fontFamily: 'theme.typography.fontFamily',
 
We should not hard code font size, instead use them variants
We should not give font family explicitly.

**Defining Constants**
const styles = {} 
const ShipItemList
const PickupItemList

should be defined outside component


**Using localization**
should not hard code text, instead use localization
'Shipping to Home'
'Select Shipping Option'
'Pick up in Store'


**'Select Shipping Option' dropdown**
'Select Shipping Option' is place holder, user should not able to select it.

File: components/checkout/Shipping/OrderItems/ShippingMethod.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  pickupItems: ProductItemProps[]
+  orderShipmentMethods: ShippingRate[]
+  onChange: (name: string, value: string) => void
+}
+
+const ShippingMethod = (props: ShippingMethodProps) => {
+  const { shipItems, pickupItems, orderShipmentMethods, onChange } = props
+  const { t } = useTranslation('common')
+
+  const styles = {
```
Comment (kibo-kevinwatts) [inline components/checkout/Shipping/OrderItems/ShippingMethod.tsx:31]: styles should be defined outside component

File: components/checkout/Shipping/OrderItems/ShippingMethod.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    shippingType: {
+      variant: 'subtitle1',
+      component: 'span',
+      fontWeight: '600',
+      color: 'text.primary',
+    },
+  }
+
+  const ShipItemList = () => (
```
Comment (kibo-kevinwatts) [inline components/checkout/Shipping/OrderItems/ShippingMethod.tsx:40]: shipitemlist should be defined outside component

File: components/checkout/Shipping/OrderItems/ShippingMethod.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  price?: string
+  salePrice?: string
+  children?: ReactNode
+}
+
+export type ShippingMethodProps = {
+  shipItems: ProductItemProps[]
+  pickupItems: ProductItemProps[]
+  orderShipmentMethods: ShippingRate[]
+  onChange: (name: string, value: string) => void
```
Comment (kibo-kevinwatts) [inline components/checkout/Shipping/OrderItems/ShippingMethod.tsx:24]: can we make the `onChange` more descriptive? `onShippingMethodChange` or something that says what is changing

File: components/common/KiboImage/KiboImage.tsx [hunk 8]
```diff
@@ -1,7 +1,20 @@
+import { SyntheticEvent } from 'react'
+
 import Image, { ImageProps } from 'next/image'
 
+import DefaultImage from '@/public/product_placeholder.svg'
+
+const onImageError = (
```
Comment (kibo-kevinwatts) [inline components/common/KiboImage/KiboImage.tsx:7]: KiboImage is used for more things than product images, we won't always want to replace with product placeholder on error.

File: components/checkout/Shipping/OrderItems/ShippingMethod.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    }
+    setup(params)
+
+    const shipItemTitle = screen.getByTestId('ship-title')
+    expect(shipItemTitle).toBeInTheDocument()
+
+    const pickupItemTitle = screen.getByTestId('pickup-title')
+    expect(pickupItemTitle).toBeInTheDocument()
+  })
+})
```
Comment (kibo-kevinwatts) [inline components/checkout/Shipping/OrderItems/ShippingMethod.spec.tsx:106]: Add another test case fo the shipping method being selected

File: components/checkout/Shipping/OrderItems/ShippingMethod.tsx (hunk 5 above)
Comment (sss-bct) [reply components/checkout/Shipping/OrderItems/ShippingMethod.tsx]: Fixed.

File: components/checkout/Shipping/OrderItems/ShippingMethod.tsx (hunk 6 above)
Comment (sss-bct) [reply components/checkout/Shipping/OrderItems/ShippingMethod.tsx]: Fixed.

File: components/checkout/Shipping/OrderItems/ShippingMethod.tsx (hunk 7 above)
Comment (sss-bct) [reply components/checkout/Shipping/OrderItems/ShippingMethod.tsx]: Fixed.

File: components/common/KiboImage/KiboImage.tsx (hunk 8 above)
Comment (sss-bct) [reply components/common/KiboImage/KiboImage.tsx:14]: Resolved by passing error props.

File: components/checkout/Shipping/OrderItems/ShippingMethod.spec.tsx (hunk 9 above)
Comment (sss-bct) [reply components/checkout/Shipping/OrderItems/ShippingMethod.spec.tsx:107]: That is handled in integration test. Do we need it here as well ?

File: components/common/ProductItemList/ProductItemList.tsx [hunk 10]
```diff
@@ -6,10 +6,12 @@ import ProductItem, { ProductItemProps } from '@/components/common/ProductItem/P
 
 export type ProductItemListProps = {
   items: ProductItemProps[]
+  clickChageStore?: () => void
```
Comment (kibo-kevinwatts) [inline components/common/ProductItemList/ProductItemList.tsx:9]: fix typo and prefix with 'on'

**Review by kibo-kevinwatts — APPROVED** [review]

**Review by kibo-sushant — APPROVED** [review]

File: components/common/ProductItemList/ProductItemList.tsx (hunk 10 above)
Comment (sss-bct) [reply components/common/ProductItemList/ProductItemList.tsx]: Typo and prefix fixed. This clickChangeStore handler is called by child component and as we are mocking child component in unit test cases, we can not write a unit test case here, but we have written integration test case .
